src/main/java/app/linguistai/bmvp/service/stats/UserLoggedDateService.java

```diff
@@ -0,0 +1,75 @@
+package app.linguistai.bmvp.service.stats;
+
+import app.linguistai.bmvp.exception.NotFoundException;
+import app.linguistai.bmvp.model.User;
+import app.linguistai.bmvp.model.stats.UserLoggedDate;
+import app.linguistai.bmvp.repository.IAccountRepository;
+import app.linguistai.bmvp.repository.stats.IUserLoggedDateRepository;
+import app.linguistai.bmvp.response.stats.RUserLoggedDate;
+import app.linguistai.bmvp.utils.DateUtils;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.sql.Date;
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+@Service
+public class UserLoggedDateService {
+
+	private final IUserLoggedDateRepository userLoggedDateRepository;
+	private final IAccountRepository accountRepository;
+
+	public RUserLoggedDate getLoggedDates(String email, String sort, Integer numDays) {
+		// Validate numDays
+		if (numDays != null && numDays < 0) {
+			throw new IllegalArgumentException("Number of days must be a non-negative integer");
+		}
+
+		// Calculate startDate if numDays is not null
+		Date startDate = null;
+		if (numDays != null) {
+			startDate = Date.valueOf(LocalDate.now().minusDays(numDays));
+		}
+
+		// Retrieve user logged dates based on sort order and startDate
+		Optional<List<UserLoggedDate>> optionalUserLoggedDates;
```

Comment:
`Optional` can be removed after `Optional` is also removed from the `JPA repository` for methods with `Lists`.